### PR TITLE
feat: support additional YAML files

### LINKED ISSUE #120: Please make the project exposes multiple YAML files to GraphQL nodes.
**What would you like to be added**:

Parameterize [typeName of gatsby-transformer-yaml plugin](https://github.com/jpedroschmitz/rocketdocs/blob/main/%40rocketseat/gatsby-theme-docs-core/gatsby-config.js?fbclid=IwAR00cTjXBC648-nY4MEUTfPme4L69h5OYlCaM3vvoRN202I3PbVH_f_8zzM#L30) or make it flexible that we can expose new YAML to new GraphQL node.

**Why is this needed**:

Since we set  typeName to static value `SidebarItems`, we cannot expose new YAML file to a new GraphQL node.

**Is your enhancement request related to a problem? Please describe.**

I would like to add new data stored in YAML and expose it as GraphQL node that I can query later.
However, adding `gatsby-transformer-yaml` plugin section in my custom gatsby-config doesn't help.
This is because it cannot override the existing ` typeName: 'SidebarItems'` and all new YML files will be merged to this type.
This also causes  an issue when trying to [query allSidebarItems in gast-by-node.js].(https://github.com/jpedroschmitz/rocketdocs/blob/main/%40rocketseat/gatsby-theme-docs-core/gatsby-node.js#L75)

**Additional context**

This screenshot took from my personal project that I added multiple YAML files and exposed them to GraphQL nodes.
I only use default type name of [gatsby-transformer-yaml](https://github.com/dotnetthailand/dotnetthailand.github.io/blob/main/gatsby-config.js#L120)

image

### PR DESCRIPTION
**Pull request by aaronamm**
**Changes proposed**

- The main change of this PR is setting gatsby-transformer-yaml's option to support additional YAML files.
  - If the current processed file is `sidebar.yml`, expose as  `sidebarSidebarItems` node and keep the current behavior, no breaking change if other people are about to update a newer version with this merged PR.
- Add `dataPath` configuration and set a default value to `data` folder's name.
- Add `using-yaml-files.md` of how to use YAML files in the project.
- Rename imported SEO component to Seo to fix Pascal case naming warning.

My concern
- If a user update gatsby-theme-docs-core to version 3.1.0 (minor), they will get a data folder automatically. 
  However if one does not commit a data folder to a repository because it is an empty folder, this causes [gatsby-source-filesystem throw an error](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-source-filesystem/src/gatsby-node.js#L169)
- I think after this PR get merged, we need to update `@rocketseat/gatsby-theme-docs` to use `@rocketseat/gatsby-theme-docs-core@3.1.0`.  However, I get confused because I am very new with changeset tool and don't know will it be patch or minor for a version of `@rocketseat/gatsby-theme-docs` project. Therefore, I leave this for you. 
- If you need me to do an addition PR to update `@rocketseat/gatsby-theme-docs`'s version, please let me know if it will be patch or minor because I don't know if you want to keep number versions of these these two projects synced.
- We should consider to not specify typeName of sidebar.yml file and use a default typeName of `gatsby-transformer-yaml`.
  This is because we can remove our custom typeName function and simplify our code. However, this causes a breaking change and is not backward compatibility for existing projects that use @rocketseat.

Fixes #120 

**Additional context**

N/A

### CODE REVIEW

**Comment by aaronamm** [comment]
@jpedroschmitz Is it possible for you to review this PR, I kind of need this feature. Thank you so much.

**Comment by jpedroschmitz** [comment]
Hey @aaronamm, sorry for taking so long to review it. A lot of stuff is going on right now.

I haven't worked with Gatsby in a long time, so I'm not sure about the code itself.

About the data folder, maybe we should rename it to `yamlFiles` or something like that. I think `data` is confusing.

--

The changeset will manage all the dependencies updates around the project! And minor is fine =D

**Comment by aaronamm** [comment]
@jpedroschmitz Thanks for your code review and sorry for late reply. I have changed "data" folder to "yamlFiles" already.
However, I am very new for changeset tool. I made a [new commit ](https://github.com/jpedroschmitz/rocketdocs/pull/124/commits/98c2100f312542dc7ba053148e1ac661c222204d) as a refactor commit without a new changeset file.

I tested this change and it works without any issues. If everything is okay, please could you merge this PR.
Thanks.

**Review by jpedroschmitz — APPROVED** [review]
Thanks a lot for working on this @aaronamm!

I did just a few small changes, but the rest is looking good!

And also thanks for adding docs for the new feature! Really important =D

File: examples/gatsby-theme-docs/gatsby-config.js [hunk 1]
```diff
@@ -11,7 +11,6 @@ module.exports = {
     themeColor: `#8257E6`,
     basePath: `/`,
   },
-  flags: { PRESERVE_WEBPACK_CACHE: true },
```
Comment (aaronamm) [inline examples/gatsby-theme-docs/gatsby-config.js:14]: This is option useful for speed up compilation steps. Please let me know why did you remove it. Thanks.

**Comment by aaronamm** [comment]
@jpedroschmitz  Thank you so much for merging this PR and help me improve the code. 
However, is it possible for you to answer my question a last comment.
Thanks again.

File: examples/gatsby-theme-docs/gatsby-config.js (hunk 1 above)
Comment (jpedroschmitz) [reply examples/gatsby-theme-docs/gatsby-config.js:14]: It's no longer required on Gatsby 4, as it's the default. There was a warning about it.

**Comment by jpedroschmitz** [comment]
I still couldn't publish it because my NPM token is no longer valid :/

Since I changed companies, I'm no longer in Rocketseat NPM organization.

I'll try to get a new one or rename the package.

File: examples/gatsby-theme-docs/gatsby-config.js (hunk 1 above)
Comment (aaronamm) [reply examples/gatsby-theme-docs/gatsby-config.js:14]: @jpedroschmitz  Nice, thanks you so much for letting me know. I should have updated with Gatsby 4. ^^.

**Comment by aaronamm** [comment]
@jpedroschmitz  Thank you for you supporting. I appreciate your help.

**Comment by jpedroschmitz** [comment]
Hey, I was able to get a new token! The release was published!

Thanks for your patience ❤️

**Comment by aaronamm** [comment]
@jpedroschmitz  Thank you so much.
